frontend/src/components/teacher/view-sessions/TestSessionListItemPopover.tsx

```diff
@@ -38,12 +41,40 @@ const TestSessionListItemPopover = ({
     history.push(Routes.DISTRIBUTE_ASSESSMENT_PAGE, session);
   };
 
+  const [previewTest] = useLazyQuery<{
+    test: TestResponse;
+  }>(GET_TEST);
+
+  const onPreviewTest = async () => {
+    const { data } = await previewTest({
+      variables: { id: session.testId },
+    });
+    if (data) {
+      history.push({
+        pathname: Routes.TEACHER_ASSESSMENT_PREVIEW_PAGE({
+          assessmentId: session.testId,
+        }),
+        state: data.test,
+      });
+    } else {
+      showToast({
+        message:
+          "This assessment cannot be previewed at this time. Please try again.",
+        status: "error",
+      });
+    }
+  };
+
   return (
     <Popover>
       <VStack divider={<Divider />} spacing={0}>
         <PopoverButton name="Edit" onClick={onEditTestSession} />
         {session.status !== TestSessionStatus.PAST && (
-          <PopoverButton name="Delete" onClick={openDeleteModal} />
+          <>
+            <PopoverButton name="Delete" onClick={openDeleteModal} />
+            <Divider />
```

Comment:
Do we need the extra divider here? I think `VStack` with the `divider` prop should automatically handle it, provided you remove the extra fragment (`<>`). (If we want to hide the whole popover for past sessions, it would make more sense IMO to either conditionally return null from this component, or (better) conditionally render the entire `TestSessionListItemPopover` component in the parent `TestSessionListItem` or wherever else this component is used.)